src/System/Relay/middleware/__tests__/cacheHeaderMiddleware.jest.ts

```diff
@@ -52,95 +52,127 @@ describe("cacheHeaderMiddleware", () => {
   })
 
   describe("Cache-Control headers", () => {
-    it("skips CDN cache if logged in", async () => {
-      jest.spyOn(window as any, "location", "get").mockImplementation(() => ({
-        pathname: "/not-found",
-      }))
+    describe("allowing a request to be cached by the CDN", () => {
+      it("allows caching even when logged in if the @cacheable directive was used", async () => {
+        req.operation = {
+          text: `
+          query ArtistQuery @cacheable {
+            artist(id: "andy-warhol) {
+              name
+            }
+          }
+        `,
+        }
+
+        jest.spyOn(window as any, "location", "get").mockImplementation(() => ({
+          pathname: "/artists",
+        }))
+
+        next.mockResolvedValue({ status: 200 })
+
+        const middleware = cacheHeaderMiddleware({
+          user: { email: "foo" },
+          url: "/foo",
+        })(next)
+        const res = await middleware(req)
+
+        expect(req.fetchOpts.headers["Cache-Control"]).toBeUndefined()
+        expect(next).toHaveBeenCalledWith(req)
+        expect(res).toEqual({ status: 200 })
+      })
+    })
 
-      next.mockResolvedValue({ status: 200 })
+    describe("skipping the cache", () => {
```

Comment:
This is just a noisy diff as i moved a bunch of 'skip' tests into their own describe block